tests/integration/test_types_integration.py

```diff
@@ -569,133 +579,129 @@ def query_timestamp_with_timezone(trino_connection, tz_str):
+            python=_localize(datetime(2001, 8, 22, 0, 0, 0, 0), tz_str))
         .add_field(
             sql=f"TIMESTAMP '2001-08-22 00:00:00.00000050 {tz_str}'",
-            python=datetime(2001, 8, 22, 0, 0, 0, 0).replace(tzinfo=tz))
+            python=_localize(datetime(2001, 8, 22, 0, 0, 0, 0), tz_str))
         .add_field(
             sql=f"TIMESTAMP '2001-08-22 23:59:59.9999994 {tz_str}'",
-            python=datetime(2001, 8, 22, 23, 59, 59, 999999).replace(tzinfo=tz))
+            python=_localize(datetime(2001, 8, 22, 23, 59, 59, 999999), tz_str))
         .add_field(
             sql=f"TIMESTAMP '2001-08-22 23:59:59.9999994999 {tz_str}'",
-            python=datetime(2001, 8, 22, 23, 59, 59, 999999).replace(tzinfo=tz))
+            python=_localize(datetime(2001, 8, 22, 23, 59, 59, 999999), tz_str))
         # round up
         .add_field(
             sql=f"TIMESTAMP '2001-08-22 12:34:56.123456789 {tz_str}'",
-            python=datetime(2001, 8, 22, 12, 34, 56, 123457).replace(tzinfo=tz))
+            python=_localize(datetime(2001, 8, 22, 12, 34, 56, 123457), tz_str))
         # round up, min value
         .add_field(
             sql=f"TIMESTAMP '2001-08-22 00:00:00.000000500001 {tz_str}'",
-            python=datetime(2001, 8, 22, 0, 0, 0, 1).replace(tzinfo=tz))
+            python=_localize(datetime(2001, 8, 22, 0, 0, 0, 1), tz_str))
         # round up, max value
         .add_field(
             sql=f"TIMESTAMP '2001-08-22 00:00:00.0000009 {tz_str}'",
-            python=datetime(2001, 8, 22, 0, 0, 0, 1).replace(tzinfo=tz))
+            python=_localize(datetime(2001, 8, 22, 0, 0, 0, 1), tz_str))
         .add_field(
             sql=f"TIMESTAMP '2001-08-22 00:00:00.00000099999 {tz_str}'",
-            python=datetime(2001, 8, 22, 0, 0, 0, 1).replace(tzinfo=tz))
+            python=_localize(datetime(2001, 8, 22, 0, 0, 0, 1), tz_str))
         # round up to next day, min value
         .add_field(
             sql=f"TIMESTAMP '2001-08-22 23:59:59.9999995 {tz_str}'",
-            python=datetime(2001, 8, 23, 0, 0, 0, 0).replace(tzinfo=tz))
+            python=_localize(datetime(2001, 8, 23, 0, 0, 0, 0), tz_str))
         .add_field(
             sql=f"TIMESTAMP '2001-08-22 23:59:59.999999500001 {tz_str}'",
-            python=datetime(2001, 8, 23, 0, 0, 0, 0).replace(tzinfo=tz))
+            python=_localize(datetime(2001, 8, 23, 0, 0, 0, 0), tz_str))
         # round up to next day, max value
         .add_field(
             sql=f"TIMESTAMP '2001-08-22 23:59:59.9999999 {tz_str}'",
-            python=datetime(2001, 8, 23, 0, 0, 0, 0).replace(tzinfo=tz))
+            python=_localize(datetime(2001, 8, 23, 0, 0, 0, 0), tz_str))
         .add_field(
             sql=f"TIMESTAMP '2001-08-22 23:59:59.999999999 {tz_str}'",
-            python=datetime(2001, 8, 23, 0, 0, 0, 0).replace(tzinfo=tz))
-        # ce
-        .add_field(
-            sql=f"TIMESTAMP '0001-01-01 01:23:45.123 {tz_str}'",
```

Comment:
This is a non-representative date for the US/Eastern and Asia/Kolkata time zones.